### DISCUSSION TITLE
What is included in 2.4.3-pre.11?

### DISCUSSION THREAD

**Discussion by RomanKalachik** [xunit/xunit#2708] [Question] [answered]
Downloaded the latest preview to test Explicit support in FactAttribute. For some reason this property is missing. How can one determine   what is icluded in a preview builds? Do you use git Tag feature in this project?

**Comment by bradwilson** [comment, marked as answer]
Pre-release builds come from pushes into the appropriate branch (`main` for v3, `v2` for v2) and are uploaded to MyGet. There is no tagging, as that would involve tagging every single commit. We don't write release notes for pre-release builds until we ship them to NuGet.

There are a few ways you can correlate a CI build back to a specific source commit (aka Git SHA).

## Console runner

This is probably the simplest way. If you run the console runner, it will give you the Git SHA that was used to build it inside of its version number. For example, running <b>`%USERPROFILE%\.nuget\packages\xunit.runner.console\2.4.3-pre.11\tools\net472\xunit.console.exe`</b>:

```
xUnit.net Console Runner v2.4.3-pre.11+2f51d3627a (64-bit Desktop .NET 4.7.2, runtime: 4.0.30319.42000)
Copyright (C) .NET Foundation.
[...]
```

The `2f51d3627a` portion of the version number is the Git SHA.

## Decompiler

The version number is embedded into every binary (with an assembly-level `AssemblyInformationalVersionAttribute`), so you could also use a tool like [ILSpy](https://github.com/icsharpcode/ILSpy) to see it:

![Screenshot of ILSpy showing AssemblyInformationalVersionAttribute](https://user-images.githubusercontent.com/16944/235252398-b1e74678-eb50-49e1-bbe0-309ece606e04.png)

## Find the CI build on GitHub

Use the push date on [MyGet](https://myget.org/feed/xunit/package/nuget/xunit/2.4.3-pre.11) to get the build date:

![Screenshot showing 2.4.3-pre.11 was built on April 19, 2023](https://user-images.githubusercontent.com/16944/235253876-47937103-7e0b-4982-8c1a-56cca7df20e7.png)

Then go to the GitHub Actions CI builds for the `v2` branch: https://github.com/xunit/xunit/actions/workflows/push-v2.yaml

Unfortunately the "friendly" date display (aka, "last week") is a bit inaccurate for our needs, but in this case there's only one candidate:

![Screenshot of v2 CI builds](https://user-images.githubusercontent.com/16944/235254215-64950a6f-2c7c-4f12-aeae-66998165e51d.png)

Obviously we already know that's correct (since we already know the SHA 😁), but if you have multiple candidates, you can [click into the CI build](https://github.com/xunit/xunit/actions/runs/4747358346/jobs/8432165542), and search for your version to make sure you have the correct CI build, and that'll show the console runner header:

![Screenshot of the CI build with a version search, showing a hit for the console runner](https://user-images.githubusercontent.com/16944/235255134-5c4f6291-b023-4d2f-bce8-c1f2eb06fdd8.png)
